jsapp/js/mixins.tsx

```diff
@@ -35,63 +36,172 @@ import {
   notify,
   escapeHtml,
   buildUserUrl,
-  renderCheckbox
-} from 'utils';
+  renderCheckbox,
+} from 'js/utils';
 import myLibraryStore from 'js/components/library/myLibraryStore';
+import type {
+  CreateImportRequest,
+  ImportResponse,
+  Permission,
+  SubmissionResponse,
+} from 'js/dataInterface';
 
 const IMPORT_CHECK_INTERVAL = 1000;
 
-var mixins = {};
+interface MixinsObject {
+  contextRouter: {
+    [functionName: string]: Function;
+    context?: any; // Hack to not go into typing the mixins too much
```

Comment:
I would say any usage of `any` is a "hack" we want to remove. Might be better to comment the rare, legitimate usage of any and assume other usages are temporary hacks to remove when possible.